crates/ty_python_semantic/src/types.rs

```diff
@@ -5766,11 +5723,15 @@ pub enum KnownInstanceType<'db> {
 }
 
 impl<'db> KnownInstanceType<'db> {
-    fn normalized(self, db: &'db dyn Db) -> Self {
+    fn normalized_impl(self, db: &'db dyn Db, seen_types: &mut SeenTypes<'db>) -> Self {
         match self {
-            Self::SubscriptedProtocol(context) => Self::SubscriptedProtocol(context.normalized(db)),
-            Self::SubscriptedGeneric(context) => Self::SubscriptedGeneric(context.normalized(db)),
-            Self::TypeVar(typevar) => Self::TypeVar(typevar.normalized(db)),
+            Self::SubscriptedProtocol(context) => {
+                Self::SubscriptedProtocol(context.normalized_impl(db, seen_types))
+            }
+            Self::SubscriptedGeneric(context) => {
+                Self::SubscriptedGeneric(context.normalized_impl(db, seen_types))
+            }
+            Self::TypeVar(typevar) => Self::TypeVar(typevar.normalized_impl(db, seen_types)),
             Self::TypeAliasType(type_alias) => Self::TypeAliasType(type_alias.normalized(db)),
```

Comment:
do we also need to create a `TypeAliasType::normalized_impl` method that passes down `seen_types`?

Reply:
Yes, good catch